backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/Log.java

```diff
@@ -810,4 +814,30 @@ SearchException cannotHighlightInNestedContext(String currentNestingField,
 	@Message(id = ID_OFFSET + 172, value = "'%1$s' cannot be nested in an object projection. "
 			+ "%2$s")
 	SearchException cannotUseProjectionInNestedContext(String projection, String hint, @Param EventContext eventContext);
+
+	@Message(id = ID_OFFSET + 173, value = "The targeted Elasticsearch cluster is reachable, but does not expose its version."
+			+ " Check that the configured Elasticsearch hosts/URI points to the right server."
+			+ " If you are targeting Amazon OpenSearch Serverless, you must set the configuration property '%1$s' explicitly to '%2$s'."
+			+ " See the reference documentation for more information.")
+	SearchException unableToFetchElasticsearchVersion(String versionConfigPropertyKey,
+			ElasticsearchVersion expectedAWSOpenSearchServerlessVersion);
+
+	@Message(id = ID_OFFSET + 174, value = "Cannot check the Elasticsearch version because the targeted Elasticsearch distribution '%s' does not expose its version.")
+	SearchException cannotCheckElasticsearchVersion(ElasticsearchDistributionName distributionName);
+
+	@Message(id = ID_OFFSET + 175,
+			value = "Unexpected Amazon OpenSearch Serverless version: '%1$s'."
+					+ " Amazon OpenSearch Serverless doesn't use version numbers."
+					+ " Set the version to simply '%2$s'.")
+	SearchException unexpectedAwsOpenSearchServerlessVersion(ElasticsearchVersion configuredVersion,
+			ElasticsearchVersion expectedAWSOpenSearchServerlessVersion);
+
+	@Message(id = ID_OFFSET + 176, value = "Cannot execute '%s' because Amazon OpenSearch Serverless does not support this operation."
+			+ " Either avoid this operation or switch to another Elasticsearch/OpenSearch distribution.")
+	SearchException cannotExecuteOperationOnAmazonOpenSearchServerless(String flush);
```

Comment:
I tried to limit the impact of those changes, because I wouldn't want to change e.g. the mapper just because of this incubating feature.

Anyway, in the case of index open/close, the operations are executed conditionally, only if there are settings changes, so we can't really fail early.

Thanks for the suggestion, will apply.